packages/react-dom/src/__tests__/ReactDOMInput-test.js

```diff
@@ -1002,29 +1002,6 @@ describe('ReactDOMInput', () => {
     expect(node.value).toBe('0');
   });
 
-  it('should properly control 0.0 for a number input', async () => {
```

Comment:
Why remove these tests? The new tests is tied to Form Actions so it would make sense to me to keep these tests.

Reply:
Yes, I thought for a long about what to do with this test and left it as text in the PR description:

> In doing so, I removed the test "`should properly control 0.0 for a number input`". I don't understand why it even exists - why is it expected that a controlled field can be overwritten in this way? It feels like it was working by sheer miracle. But we can bring back the `ChangeEventPlugin` logic. However, as I've come to understand, it's also full of problems and seems quite prone to bugs. Plus, it's unnecessary actions

It turns out this test is no longer relevant, as the value doesn't change on blur anymore. Moreover, I don't understand why it should have changed in a controlled field in this way? It doesn't change like this anywhere else...

Reply:
> I don't understand why it even exists - why is it expected that a controlled field can be overwritten in this way?

It seems to me it's just testing that you can input "0.0" which is a valid numeric input. Maybe I'm just not understanding what the new behavior is. Let's start by restoring the test and just updating the behavior.

These tests should include the actual user behavior we're simulating so that we can verify that the actual behavior is correct and it's just the test that needs updating. Can you do that for the new tests?

Reply:
There's a very strange situation with the tests here. It turns out I removed the logic that set the defaultValue for number input on blur.

The check in this test is as follows - render with a controllable value of 0, focus, change the value to 0.0, blur.

After I removed the blur logic, this test failed. Which seems logical, but why does the logic of setting defaultValue play any role in this case? The field had a value of 0, how did changing defaultValue affect this value?

How should this work in real experience? That the user changed the value, then they left the field and it became 0.0?

Reply:
It's very unfortunate that we can't conduct a normal test specifically in the browser env. I encountered this problem in submit tests and here it's similar. _Just out of curiosity, do you have any thoughts on setting up playwright or something like it here?_

Reply:
Would it be sufficient to simply have a test where an input is rendered with a value of 0.0 and it's immediately checked that its value is indeed 0.0?

This sounds strange, but it's the only logical part in this test

Reply:
> Just out of curiosity, do you have any thoughts on setting up playwright or something like it here?

Yeah but that's out of scope here. We can use Codesandboxes and Codesandbox Preview builds to verify the behavior.

It'd still be valueable for now to restore the test and change the assertions. We're currently shipping breaking changes so a subtle change to how DOM event listeners may observe events may be fine.